tools/nightly-builder/latest-build.js

```diff
@@ -0,0 +1,36 @@
+"use strict"
+
+const jsonist = require('jsonist')
+
+const indexUrl = 'https://iojs.org/download/{type}/index.json'
+
+
+function latestBuild (type, callback) {
+  let url = indexUrl.replace(/\{type\}/, type)
+
+  function onGet (err, data) {
+    if (err)
+      return callback(err)
+
+    let i = -1
+
+    while (data[++i]) {
+      let m      = data[i].version.match(/nightly(20\d\d)(\d\d)(\d\d)(\w{10,})/)
+        , date   = new Date(m && `${m[1]}-${m[2]}-${m[3]}` || data[i].date)
+        , commit = m && m[4]
+
+      return callback(null, {
+          version : data[i].version
+        , date    : date
+        , commit  : commit
+      })
+    }
+
+    return callback(new Error(`no latest version for "${type}"`))
```

Comment:
More concise to bail to this before the above if there isn't data. i.e:

`if (!data[0]) return callback(new Error(...))`